File: 88. Merge Sorted Array/Merge_Sorted_Array.java
public class Merge_Sorted_Array {
    
     static void mergeFunction (int x, int[] nums1,int m) {
        boolean isChange = false;
        for (int i = 0; i < m; i++) {
            if (nums1[i] > x) {
                isChange = true;
                for (int k = m-1; k >= i; k--) {
                   nums1[k+1] = nums1[k];
                }
                nums1[i] = x;
                break;
            }
        }  
        if (isChange == false) {
                nums1[m] = x;
            }
    }

        static  void merge(int[] nums1, int m, int[] nums2, int n) {
        for (int a : nums2) {
            mergeFunction(a,nums1,m);
            m++;
        }
        for (int x : nums1) {
            System.out.print(x + " ");
        }
    }

    public static void main(String[] args) {
        int[] nums1 = {1,2,3,0,0,0};
        int[] nums2 = {2,5,6};
        int m = 3;
        int n = 3;
        merge(nums1, m, nums2, n);
    }

}

File: 88. Merge Sorted Array/Merge_Sorted_Array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        p1=m-1
        p2=n-1
        p3=m+n-1
        while p1 >= 0 and p2 >= 0:
            if (nums1[p1] > nums2[p2]):
                nums1[p3] = nums1[p1]
                p1-=1
            else:
                nums1[p3] = nums2[p2]
                p2-=1
            p3-=1
        while p2 >= 0:
            nums1[p3] = nums2[p2]
            p2-=1
            p3-=1
                
           
nums1 = [1,2,3,0,0,0]
m = 3
nums2 = [2,5,6]
n = 3
Solution().merge(nums1,m,nums2,n)
print(nums1)
